keras/src/backend/config.py

```diff
@@ -403,3 +417,35 @@ def max_steps_per_epoch():
             `None`, no limit is applied.
     """
     return _MAX_STEPS_PER_EPOCH
+
+
+if not os.path.exists(_KERAS_DIR):
+    try:
+        os.makedirs(_KERAS_DIR)
+    except OSError:
+        # Except permission denied and potential race conditions
+        pass
+
+if not os.path.exists(_config_path):
+    _config_to_save = {
+        "floatx": floatx(),
+        "epsilon": epsilon(),
+        "backend": _BACKEND,  # Use the final _BACKEND value
```

Comment:
why do we need this change? Before, we would save this default keras.json before reading the `KERAS_BACKEND` variable (default is always tensorflow). Now, we will save this default keras.json after reading `KERAS_BACKEND`, so the default might be Jax or Pytorch depending on the first Keras program you run.

This seems ripe to cause some hard to track down bugs, and has nothing to do with the other changes here right? Let's move this block back up so we can preserve the original behavior.

Reply:
Used is_nnx_enabled() in the config save logic.
Moved the config file save logic to before environment variable reading.